cloud/rackspace/rax_clb_ssl.py

```diff
@@ -0,0 +1,269 @@
+    certificate: "{{ lookup('file', 'credentials/server.crt' ) }}"
+    intermediate_certificate: "{{ lookup('file', 'credentials/trust-chain.crt') }}"
+    secure_traffic_only: true
+    wait: true
+
+- name: Disable SSL termination
+  rax_clb_ssl:
+    loadbalancer: "{{ registered_lb.balancer.id }}"
+    state: absent
+    wait: true
+'''
+
+try:
+    import pyrax
+    HAS_PYRAX = True
+except ImportError:
+    HAS_PYRAX = False
+
+def cloud_load_balancer_ssl(module, loadbalancer, state, enabled, private_key,
+                            certificate, intermediate_certificate, secure_port,
+                            secure_traffic_only, https_redirect,
+                            wait, wait_timeout):
+    # Validate arguments.
+
+    if state == 'present':
+        if not private_key:
+            module.fail_json(msg="private_key must be provided.")
+        else:
+            private_key = private_key.strip()
+
+        if not certificate:
+            module.fail_json(msg="certificate must be provided.")
+        else:
+            certificate = certificate.strip()
+
+    attempts = wait_timeout / 5
+
+    # Locate the load balancer.
+
+    balancer = rax_find_loadbalancer(module, pyrax, loadbalancer)
+    existing_ssl = balancer.get_ssl_termination()
+
+    changed = False
+
+    if state == 'present':
+        # Apply or reconfigure SSL termination on the load balancer.
+        ssl_attrs = dict(
+            securePort=secure_port,
+            privatekey=private_key,
+            certificate=certificate,
+            intermediateCertificate=intermediate_certificate,
+            enabled=enabled,
+            secureTrafficOnly=secure_traffic_only
+        )
+
+        needs_change = False
+
+        if existing_ssl:
+            for ssl_attr, value in ssl_attrs.iteritems():
+                if ssl_attr == 'privatekey':
+                    # The private key is not included in get_ssl_termination's
+                    # output (as it shouldn't be). Also, if you're changing the
+                    # private key, you'll also be changing the certificate,
+                    # so we don't lose anything by not checking it.
+                    continue
+
+                if value is not None and existing_ssl.get(ssl_attr) != value:
+                    # module.fail_json(msg='Unnecessary change', attr=ssl_attr, value=value, existing=existing_ssl.get(ssl_attr))
+                    needs_change = True
+        else:
+            needs_change = True
+
+        if needs_change:
+            try:
+                balancer.add_ssl_termination(**ssl_attrs)
+            except pyrax.exceptions.PyraxException, e:
+                module.fail_json(msg='%s' % e.message)
+            changed = True
+    elif state == 'absent':
+        # Remove SSL termination if it's already configured.
+        if existing_ssl:
+            try:
+                balancer.delete_ssl_termination()
+            except pyrax.exceptions.PyraxException, e:
```

Comment:
I noticed that `rax_clb` blanket-catches `Exception`. Is catching the somewhat-more-specific `PyraxException` instead okay?

:thought_balloon: Actually, it might be nice to have better behavior for handling uncaught exceptions from Python modules in general, maybe in a parent context rather than relying on each module to catch its own... using `module.fail_json()` strips out a traceback, for example.